src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,142 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
+    private static class Node<T> {
+        private T value;
+        private Node<T> next;
+        private Node<T> previous;
+
+        public Node(T value, Node<T> next, Node<T> previous) {
+            this.value = value;
+            this.next = next;
+            this.previous = previous;
+        }
+    }
+
+    private Node<T> getNodeByIndex(int index) {
+        Node<T> currentNode = head;
+        for (int i = 0; i < index; i++) {
+            currentNode = currentNode.next;
+        }
+        return currentNode;
+    }
+
     @Override
     public void add(T value) {
+        if (head == null) {
+            head = tail = new Node<>(value, null, null);
+        } else {
+            Node<T> newNode = new Node<>(value, null, tail);
+            tail.next = newNode;
+            tail = tail.next;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index > size || index < 0) {
```

Comment:
beter create method checkElementIndex(int index). You use this if statement - 4 times